frontend/packages/dev-console/src/components/import/devfile/devfileHooks.ts

```diff
@@ -21,24 +24,85 @@ export const useDevfileServer = (
 
   const {
     name,
-    git: { url, ref, dir },
+    git: { url, ref, dir, type, secretResource },
     devfile,
   } = values;
   const smartSlashDir = prefixDotSlash(suffixSlash(dir));
 
   const { devfileContent, devfilePath } = devfile || {};
 
-  const devfileData = React.useMemo(() => {
+  const getResourceContent = React.useCallback(
+    async (resourceURI: string) => {
+      let resourceContent;
+      if (resourceURI.startsWith('http://') || resourceURI.startsWith('https://')) {
+        try {
+          resourceContent = await coFetch(resourceURI);
+        } catch (e) {
+          // eslint-disable-next-line no-console
+          console.error(e);
+        }
```

Comment:
So in case of external URIs there is going to be CORS issue. Do we show that to the user?

Reply:
Also if we directly do fetch and if we assume there is no CORS issue then the source of the page is returned as a response if the external URI is something like 'https://github.com/devfile-samples/devfile-sample-code-with-quarkus/blob/main/devfile.yaml'.

Reply:
Any suggestions on how do we handle this?

Reply:
@jerolimov ^^

Reply:
Added a common error message.


https://user-images.githubusercontent.com/20724543/196457700-c00eb245-420d-4a61-b7f1-64734e40b516.mov

Reply:
Hey @divyanshiGupta,

> So in case of external URIs there is going to be CORS issue at the moment. Do we show that to the user?

Yes, we should show all failures that will result in a broken import to the user. I don't believe we can detect CORS issue, but we can show "Could not fetch..." in this case.

> Also if we directly do fetch and if we assume there is no CORS issue then the source of the page is returned as a response if the external URI is something like 'https://github.com/devfile-samples/devfile-sample-code-with-quarkus/blob/main/devfile.yaml'.

These GitHub frontend URLs might return CORS issues, but they also return the full HTML. You need to use a raw link instead, that also just returns the pure YAML, like: https://raw.githubusercontent.com/devfile-samples/devfile-sample-code-with-quarkus/main/devfile.yaml

I cloned the Go sample you can use that reference its own k8s resource with an absolute GitHub raw URL:

https://github.com/jerolimov/devfile-sample-go-basic-absolute-k8s-uri